packages/office-addin-debugging/src/commands.ts

```diff
@@ -1,121 +1,123 @@
-            enableSideload,
-            openDevTools,
-            document
-        });
-
-        usageDataObject.reportSuccess("start");
-    } catch (err) {
-        usageDataObject.reportException("start", err);
-        logErrorMessage(`Unable to start debugging.\n${err}`);
-    }
-}
-
-export async function stop(manifestPath: string, platform: string | undefined, command: commander.Command) {
-    try {
-        const appPlatformToDebug: Platform | undefined = parsePlatform(platform || process.env.npm_package_config_app_plaform_to_debug || Platform.Win32);
-        const dev: boolean = command.prod ? false : true;
-
-        if (manifestPath === "" && appPlatformToDebug !== undefined) {
-            manifestPath = determineManifestPath(appPlatformToDebug, dev);
-        }
-
-        await stopDebugging(manifestPath);
-        usageDataObject.reportSuccess("stop");
-    } catch (err) {
-        usageDataObject.reportException("stop", err);
-        logErrorMessage(`Unable to stop debugging.\n${err}`);
-    }
-}
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as commander from "commander";
+import * as fs from "fs";
+import { logErrorMessage, parseNumber } from "office-addin-cli";
+import * as devSettings from "office-addin-dev-settings";
+import { OfficeApp, parseOfficeApp } from "office-addin-manifest";
+import { AppType, parseDebuggingMethod, parsePlatform, Platform, startDebugging } from "./start";
+import { stopDebugging } from "./stop";
+import { usageDataObject } from './defaults';
+import { ExpectedError } from "office-addin-usage-data";
+const readJson = require('read-package-json-fast')
+
+function determineManifestPath(platform: Platform, dev: boolean): string {
+    let manifestPath = process.env.npm_package_config_manifest_location || "";
+    manifestPath = manifestPath.replace("${flavor}", dev ? "dev" : "prod").replace("${platform}", platform);
+
+    if (!manifestPath) {
+        throw new ExpectedError(`The manifest path was not provided.`);
+    }
+    if (!fs.existsSync(manifestPath)) {
+        throw new ExpectedError(`The manifest path does not exist: ${manifestPath}.`);
+    }
+
+    return manifestPath;
+}
+
+function parseDevServerPort(optionValue: any): number | undefined {
+    const devServerPort = parseNumber(optionValue, "--dev-server-port should specify a number.");
+
+    if (devServerPort !== undefined) {
+        if (!Number.isInteger(devServerPort)) {
+            throw new ExpectedError("--dev-server-port should be an integer.");
+        }
+        if ((devServerPort < 0) || (devServerPort > 65535)) {
+            throw new ExpectedError("--dev-server-port should be between 0 and 65535.");
+        }
+    }
+
+    return devServerPort;
+}
+
+export async function start(manifestPath: string, platform: string | undefined, command: commander.Command) {
+    try {
+        const pkg = await readJson(`${process.env.npm_package_json}`)

```

Comment:
I think you could make this work more simply and consolidate it into office-addin-cli. You can have a function which can take the string like "npm_package_config_app_type_to_debug", and if the package.json hasn't been parsed, you can parse it for the env vars that used to be set (npm_package_config_* and npm_package_script_*).

The goal here would be an existing piece of code that does `process.env.npm_packge_config_app_type_to_debug` should be replaced with `getPackageJsonConfig("app_type_to_debug") or getPackageJsonScript("dev_server").

Reply:
The config variable still work, so I just didn't touch them. The npm team said they would keep it for now for easier upgrades.
I can make this ```getPackageJsonScript``` and it should be better to make the code more scalable long term, I agree